pkg/image/admission/imagepolicy/imagepolicy.go

```diff
@@ -0,0 +1,392 @@
+package imagepolicy
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/golang/glog"
+	lru "github.com/hashicorp/golang-lru"
+
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	apierrs "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	"github.com/openshift/origin/pkg/api/meta"
+	"github.com/openshift/origin/pkg/client"
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	configlatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	"github.com/openshift/origin/pkg/image/admission/imagepolicy/api"
+	"github.com/openshift/origin/pkg/image/admission/imagepolicy/api/validation"
+	"github.com/openshift/origin/pkg/image/admission/imagepolicy/rules"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+func init() {
+	admission.RegisterPlugin(api.PluginName, func(client clientset.Interface, input io.Reader) (admission.Interface, error) {
+		obj, err := configlatest.ReadYAML(input)
+		if err != nil {
+			return nil, err
+		}
+		if obj == nil {
+			return nil, nil
+		}
+		config, ok := obj.(*api.ImagePolicyConfig)
+		if !ok {
+			return nil, fmt.Errorf("unexpected config object: %#v", obj)
+		}
+		if errs := validation.Validate(config); len(errs) > 0 {
+			return nil, errs.ToAggregate()
+		}
+		glog.V(5).Infof("%s admission controller loaded with config: %#v", api.PluginName, config)
+		return newImagePolicyPlugin(client, config)
+	})
+}
+
+type imagePolicyPlugin struct {
+	*admission.Handler
+	config *api.ImagePolicyConfig
+	client client.Interface
+
+	accepter rules.Accepter
+	adjuster rules.Adjuster
```

Comment:
remove and the things that implement it so they'll get reviewed in the next pull.